ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -1032,6 +1041,11 @@ func (r *RayJobReconciler) checkSubmitterAndUpdateStatusIfNeeded(ctx context.Con
 
 		if headPod == nil {
 			logger.Info("Ray head pod not found, skipping sidecar container status check")
+			// If head pod is deleted, mark the RayJob as failed
```

Comment:
If headpod == nil, is it always the case that RayJob is deleted? What about right after RayJob is created before head pod is created?